### PR TITLE
java-pkg-simple.eclass: improve MANIFEST.MF manipulation

### PR DESCRIPTION
**Pull request by vaukai**
Closes: https://bugs.gentoo.org/874924

Signed-off-by: Volkmar W. Pogatzki <gentoo@pogatzki.net>

### CODE REVIEW

**Comment by vaukai** [comment]
> i tested it on `icu4j` and got many warnings, is that ok?
> 
Thanks for testing. Just reviewed the warnings and did tests myself. Found there are no duplicate lines in `MANIFEST.MF` in any of those 3 jar files. Those warnings seem to be just warnings while `jar` seems to avoid duplicate entries automatically.

But, what's bothering me is that an empty `JAVA_MAIN_CLASS=""` from the ebuild gets transferred to `MANIFEST.MF`.
```
Manifest-Version: 1.0
Created-By: 17.0.3 (Gentoo)
Main-Class: 

```
Guess we need to replace `[[ -v JAVA_MAIN_CLASS ]]`with something else?

File: eclass/java-pkg-simple.eclass [hunk 1]
```diff
@@ -415,22 +415,25 @@ java-pkg-simple_src_compile() {
...
 	jar ${jar_args} -C ${classes} . || die "jar failed"
+	if  [[ -v JAVA_AUTOMATIC_MODULE_NAME ]]; then
+		echo "Automatic-Module-Name: ${JAVA_AUTOMATIC_MODULE_NAME}" \
+			>> "${T}/add-to-MANIFEST.MF" || die "adding module name failed"
+	fi
+	if  [[ -v JAVA_MAIN_CLASS ]]; then
+		echo "Main-Class: ${JAVA_MAIN_CLASS}" \
+			>> "${T}/add-to-MANIFEST.MF" || die "adding main class failed"
+	fi
+	if [[ -e ${T}/add-to-MANIFEST.MF ]]; then
```
Comment (Flowdalic) [inline eclass/java-pkg-simple.eclass:433]: quote T and use -f:
```suggestion
	if [[ -f "${T}"/add-to-MANIFEST.MF ]]; then
```

Comment (vaukai) [reply eclass/java-pkg-simple.eclass:433]: Thanks for the proposal. But this way we would have to do it twice: once for Main-Class and once for Automatic-module-name.
I found another solution checking for non-zero length and deleting the `add-to-MANIFEST.MF` file at the end

**Comment by Flowdalic** [comment]
I think a sentence that elaborates on the motivation for switching from -v to -n would greatly improve the commit message.

**Comment by vaukai** [comment]
> Guess we need to replace `[[ -v JAVA_MAIN_CLASS ]]`with something else?

Indeed.  `[[ -n "${JAVA_MAIN_CLASS}" ]]` does it.

**Pull request closed without merging**